feat: `paymentInitiator` role in `ServiceManagerBase`

diyahir opened the pull request:
Currently, the entity responsible for posting the range payments is the `ServiceManager` Owner, but for practical purposes this responsibility will likely be delegated to a separate entity or smart contract. 

High level changes:
- creating `ServiceManagerBaseStorage.sol`
- adding `paymentInitiator` storage var to `ServiceManagerBaseStorage`
- adding ability for owner to change `paymentInitiator`
- testing changing and proper management of `paymentInitiator` role
- formatting based on forge fmt on contracts modified

diyahir committed:
feat: payment initiator integration

- creating ServiceManagerBaseStorage
- adding paymentInitiator storage var to ServiceManagerBase
- adding ability for owner to change paymentInitiator
- formatting based on forge fmt for contracts modified

diyahir committed:
fix: typos

8sunyuan commented on src/ServiceManagerBaseStorage.sol:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    // storage gap for upgradeability
+    uint256[50] private __GAP;
```
should be 49 to accomodate the extra slot used for `paymentInitiator`

diyahir replied:
fixed

8sunyuan commented on src/ServiceManagerBase.sol:
```diff
@@ -2,28 +2,30 @@
  */
-abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
+abstract contract ServiceManagerBase is
+    ServiceManagerBaseStorage,
```
Should be  `OwnableUpgradeable`, `ServiceManagerBaseStorage` otherwise it will break the storage for existing ServiceManagers for AVSs

8sunyuan replied:
And also remove `Initializable`, already inherited from `OwnableUpgradeable`

diyahir replied:
Sounds good, yeah I was confused when I saw the registry coordinator too:
```solidity
contract RegistryCoordinator is 
    EIP712, 
    Initializable, 
    Pausable,
    OwnableUpgradeable,
    RegistryCoordinatorStorage, 
    ISocketUpdater, 
    ISignatureUtils
{
```

diyahir replied:
fixed

8sunyuan replied:
Yeah I'm not exactly sure why we double imported there

8sunyuan commented on src/ServiceManagerBase.sol:
```diff
@@ -155,29 +196,30 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
+
     // storage gap for upgradeability
     // slither-disable-next-line shadowing-state
     uint256[50] private __GAP;
```
remove storage gap here

diyahir replied:
fixed

diyahir committed:
fix: remediations

- removing erroneous gap from `ServiceManagerBase`
- typo 50 to 49
- making inheritance backward compatible

diyahir committed:
Merge branch 'dev' into feat-payment-initiator

8sunyuan reviewed (approved):
LGTM appreciate the idea and implementation on this. Going to ping for another approval on this before merging.
Also want to flag that there will be another follow-up PR to this based on the `PaymentCoordinator` being renamed to `RewardsCoordinator`. WIP PR can be found here
https://github.com/Layr-Labs/eigenlayer-contracts/pull/565

wadealexc reviewed (approved):
Looks good, thanks for the PR.

In the future, please keep code changes and code formatting in separate PRs -- it's hard to review otherwise!
